src/escapei.rs

```diff
@@ -159,11 +159,11 @@ pub fn unescape(raw: &str) -> Result<Cow<str>, EscapeError> {
 /// [HTML5 escapes]: https://dev.w3.org/html5/html-author/charref
 pub fn unescape_with<'input, 'entity, F>(
     raw: &'input str,
-    resolve_entity: F,
+    mut resolve_entity: F,
 ) -> Result<Cow<'input, str>, EscapeError>
 where
     // the lifetime of the output comes from a capture or is `'static`
-    F: Fn(&str) -> Option<&'entity str>,
+    F: FnMut(&str) -> Option<&'entity str>,
```

Comment:
After consulting with the [standard library](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map) I think, it's a good idea, but that change should be done in it's own commit and have it's own changelog entry.

Reply:
It's not immediately clear why it's necessary, as the docs [say](https://doc.rust-lang.org/std/ops/trait.FnMut.html):

> Use FnMut as a bound when you want to accept a parameter of function-like type and need to call it repeatedly, while allowing it to mutate state. If you don’t want the parameter to mutate state, use [Fn](https://doc.rust-lang.org/std/ops/trait.Fn.html) as a bound; if you don’t need to call it repeatedly, use [FnOnce](https://doc.rust-lang.org/std/ops/trait.FnOnce.html).

What state do you expect the entity resolution process to mutate?  Isn't it a read-only process?

Reply:
Yes, it is just to provide flexibility and be consistent with similar std API, such as `Iterator::map`.